addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/PathTraversalScanRule.java

```diff
@@ -145,19 +145,15 @@ public class PathTraversalScanRule extends AbstractAppParamPlugin
      * Windows/Unix/Linux/etc. local directory targets and detection pattern
      */
     private static final ContentsMatcher DIR_PATTERN = new DirNamesContentsMatcher();
-    private static final String[] LOCAL_DIR_TARGETS = {
+    private static final String[] WIN_LOCAL_DIR_TARGETS = {
```

Comment:
What's the advantage of splitting if we are using the same checks for both? Did you mean to split `DirNamesContentsMatcher` as well?

Reply:
And be good to add tests to cover that case if so.

Reply:
The goal was precisely to separation payloads for different architectures for several reasons:
- Initially, there was a problem with the weak validation function of the NIX pattern and alerts for NIX were raised on payloads for the WIN architecture;
- Separating payloads will allow them to be enriched in ways peculiar to a particular architecture;
- Since DirBased checks have restrictions on what is being sent, there is a possibility that payloads for all the architectures under research will not be sent;

Further changes are also related to this logic.
Implemented tests cover application scenarios.

Reply:
- Splitting the payloads is irrelevant to that point, what addressed the FP was further validations for NIX. We are still checking for NIX when sending WIN payloads.
- Might be the case but that's not a reason to do it now.
- That was already the case, but now it might send more (since the payloads were interleaved).

If you do not plan to change the `DirNamesContentsMatcher` I don't think there's a reason to split the payloads.

Reply:
Or, if you really want to keep them separated, add a Tech check.

Reply:
I would like to leave them in a separated state if possible. What is meant by Tech check? If this is a test, what should it be aimed at?

Reply:
Similar to what's being done in line 291 and 329.

Reply:
Something like:
https://github.com/zaproxy/zap-extensions/blob/a16b30cc891110e232fe8b24273e4d0f662885d5/addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/ElmahScanRule.java#L48-L54

Reply:
It's not something like that though, the `technologies` is only available in that method.

Reply:
Okay, fair. We commented at the same time with me not realizing there were already Tech checks in the rule

Reply:
Did it.